docs/releasing.md

```diff
@@ -0,0 +1,73 @@
+# Releasing
+
+## Prerequisites
+
+### `docker`
+
+You must have docker installed.
+
+## Output
+
+### Expected artifacts
+
+1. A container image of the shared baremetal-operator
+1. A git tag
+1. A release on Github containing:
+    - A manifest file - `infrastructure-components.yaml`
+    - A metadata file - `metadata.yaml`
+    - A cluster template - `cluster-template.yaml`
```

Comment:
I don't think we should have a metadata and cluster template files for BMO, since they are not part of capm3. Having a manifest file that allows a user to easily deploy BMO (i.e. the kustomization output) could be nice to have.

Reply:
What builds that manifest file?

Reply:
we can add a Makefile target running kustomize on the config (or deploy for now) folder